python/nano/test/openvino/pytorch/test_openvino.py

```diff
@@ -366,25 +367,94 @@ def test_openvino_trace_stable_diffusion_unet(self):
 
         unet(image_latents, torch.Tensor([980]).long(), encoder_hidden_states)
         unet(image_latents2, torch.Tensor([980]).long(), encoder_hidden_states2)
-        
+
+        # Static shape
+        dynamic_axes= False
+        nano_unet = InferenceOptimizer.trace(unet, accelerator="openvino",
+                                            input_sample=input_sample,
+                                            input_names=["sample", "timestep",
+                                                        "encoder_hidden_states", "return_dict"],
+                                            output_names=["unet_output"],
+                                            dynamic_axes=dynamic_axes,
+                                            device='CPU',
+                                            )
+        nano_unet(image_latents, torch.Tensor([980]).long(), encoder_hidden_states)
+        # Inference with wrong batchsize
+        with pytest.raises(RuntimeError,
+                        match="The input blob size is not equal to the network input size"):
+            nano_unet(image_latents2, torch.Tensor([980]).long(), encoder_hidden_states2)
+
+        # Test save, load, reshape(fail) & cache
+        with tempfile.TemporaryDirectory() as tmp_dir_name:
+            InferenceOptimizer.save(nano_unet, tmp_dir_name)
+
+            # Load without shapes
+            first_load_start = datetime.utcnow()
+            new_model = InferenceOptimizer.load(tmp_dir_name, cache_dir=tmp_dir_name)
+            first_load_end = datetime.utcnow()
+
+            # Reshape using same shape, should not recompile
+            new_model.reshape("sample[2,4,8,8],encoder_hidden_states[2,12,10]")
+            reshape_end = datetime.utcnow()
+            assert (reshape_end - first_load_end).total_seconds() * 1000 < 10
+
+            new_model(image_latents, torch.Tensor([980]).long(), encoder_hidden_states)
+            # Inference with wrong batchsize
+            with pytest.raises(RuntimeError,
+                            match="The input blob size is not equal to the network input size"):
+                new_model(image_latents2, torch.Tensor([980]).long(), encoder_hidden_states2)
+
+            # Load with caching
+            second_load_start = datetime.utcnow()
+            new_model = InferenceOptimizer.load(tmp_dir_name, cache_dir=tmp_dir_name)
+            second_load_end = datetime.utcnow()
+            assert (second_load_end - second_load_start) < (first_load_end - first_load_start)
+
+            # Reshape failed, should not recompile
+            new_model.reshape("sample[1,4,64,64],encoder_hidden_states[1,12,10]")
```

Comment:
It would be better that we can report some warnings for user to remind the reshape op is failed.

Reply:
The reshape function prints error messages returned by openvino.